presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/StandardColumnMappings.java

```diff
@@ -282,11 +282,19 @@ public static ColumnMapping timeColumnMappingUsingSqlTime()
                 TIME,
                 (resultSet, columnIndex) -> {
                     Time time = resultSet.getTime(columnIndex);
-                    return time.getTime() * PICOSECONDS_PER_MILLISECOND % PICOSECONDS_PER_DAY;
+                    return (roundDiv(toLocalTime(time).toNanoOfDay(), NANOSECONDS_PER_MILLISECOND) * PICOSECONDS_PER_MILLISECOND) % PICOSECONDS_PER_DAY;
```

Comment:
Why does this translation go through milliseconds? Shouldn't this be the same?

```
toLocalTime(time).toNanoOfDay() * PICOSECONDS_PER_NANOSECOND
```

The millisecond conversion serves to truncate the time, but `toLocalTime` returns a value with nothing more precise than nanoseconds, so the truncation never does anything.